main.py

```diff
@@ -0,0 +1,69 @@
+"""Proxy server to communicate a client to ARTIQ."""
+
+from contextlib import asynccontextmanager
+import json
+import os
+
+from fastapi import FastAPI
+from sipyco.pc_rpc import Client
+
+
+settings = {}
+
+
+def load_setup_file():
+    """Loads set-up information from the setup file.
+
+    The file should have the following JSON structure:
+
+      {
+        "master_path": {master_path}
+      }
+    """
+    global settings
+    with open("setup.json", encoding="utf-8") as setup_file:
+        settings = json.load(setup_file)
+
+
+@asynccontextmanager
+async def lifespan(_app: FastAPI):
+    """Lifespan events.
+
+    This function is set as the lifespan of the application.
+
+    The code before 'yield' is executed once before the application starts.
+    The code after 'yield' is executed when the application is shutting down.
+    """
+    load_setup_file()
+    yield
+
+
+app = FastAPI(lifespan=lifespan)
+
+
+@app.get("/ls/")
+async def list_directory(directory: str = ""):
+    """Get the list of elements in the given path.
+
+    The 'master_path' is used for the prefix of the path.
+    """
+    remote = get_client("master_experiment_db")
+    contents = remote.list_directory(os.path.join(settings["master_path"], directory))
+    return contents
```

Comment:
Hmm, from this code, the return type is quite invisible.
How do we annotate such information for the FastAPI path operation functions?

Reply:
Referred to [Response Model](https://fastapi.tiangolo.com/ko/tutorial/response-model/), we can restrict the return type!
I will apply it in according to the docs.

Reply:
Wow, what a fancy feature!
However, I think the `response_model` feature is not necessary for our case, since we know that the return type is `List[str]`, and it is not even a _Pydantic model_.
Isn't it enough to use just the type annotation?

Reply:
That's right! I set up the model for future consistency, but I'll take it out if we don't need it.

Reply:
As far as I understand, we only need it when we use Pydantic models, and want to return in a different type (e.g., in `dict`).
We know that the `remote.list_directory()` returns a list of strings, so we don't need it here.
If we use a Pydantic model later, we can use it that time.